src/DominantJuice.sol

```diff
@@ -0,0 +1,396 @@
+    IJBSingleTokenPaymentTerminal public paymentTerminal;
+
+    /// Juicebox Project State Variables
+    uint256 public projectId;
+    uint256 public projectConfiguration;
+    address public paymentToken;
+
+    /// Dominant Assurance State Variables
+    uint256 public cycleExpiryDate;
+    uint256 public cycleTarget;
+    uint256 public minimumPledgeAmount;
+    uint256 public totalRefundBonus;
+    uint256 public totalAmountPledged;
+    uint256 public totalAmountPledged_fromJB;
+    UD60x18 public rateOfDecay = ud(0.75e18); // Need to confirm we want to hardcode this
+    bool public isCycleExpired = false;
+    bool public isTargetMet = false;
+    bool public hasCreatorWithdrawnAllFunds = false;
+
+    /// Pledger Mappings and Array
+    mapping(address => uint256) public pledgedAmount;
+    mapping(address => bool) public hasBeenRefunded;
+    mapping(uint256 => mapping(address => uint256)) public orderToPledgerToAmount;
+    address[] public pledgers;
+    uint32 public pledgeOrder; // make internal?
+
+    // Events for transparency to pledgers
+    event RefundBonusDeposited(address, uint256 indexed);
+    event CycleHasClosed(bool indexed, bool indexed);
+    event CycleRefundBonusWithdrawal(address indexed, uint256 indexed);
+    event OwnerWithdrawal(address, uint256);
+    event Test(string, uint256);
+
+    error DataSourceNotInitialized();
+    error FundsMustMatchInputAmount(uint256 input);
+    error AmountIsBelowMinimumPledge(uint256 minAmount);
+    error InvalidPaymentEvent(address caller, uint256 projectId, uint256 value);
+    error CycleHasExpired();
+    error FunctionHasAlreadyBeenCalled();
+    error MustBePledger();
+    error CycleHasNotEndedYet(uint256 endTimestamp);
+    error NoRefundsForSuccessfulCycle();
+    error ContractAlreadyInitialized();
+    error AlreadyWithdrawnRefund();
+    error InvalidRedemptionEvent(address caller, uint256 projectId, uint256 value);
+    error InsufficientFunds();
+
+    /// @notice Indicates if this contract adheres to the specified interface.
+    /// @dev See {IERC165-supportsInterface}.
+    /// @param _interfaceId The ID of the interface to check for adherence to.
+    /// @return A flag indicating if the provided interface ID is supported.
+    function supportsInterface(bytes4 _interfaceId) public view virtual override returns (bool) {
+        return _interfaceId == type(IJBFundingCycleDataSource3_1_1).interfaceId
+            || _interfaceId == type(IJBPayDelegate3_1_1).interfaceId
+            || _interfaceId == type(IJBRedemptionDelegate3_1_1).interfaceId;
+    }
+
+    /// @notice Initializes the Dominant Assurance cloned contract with project details
+    /// and the pertinent Juicebox architecture contracts. Function should be called by the
+    /// Delegate Deployer contract directly after it creates the clone for each new project.
+    /// @param _projectId The ID of the project this contract's functionality applies to.
+    function initialize(
```

Comment:
initialization functions are more complicated than simple constructors. Since you're not using upgradeable contracts, it's unclear why this is needed. Since the deployer has both pieces of code, there's no real reason it can't use a constructor to initialize any code that's needed during deployment.

Reply:
Yeah, I think we can go back to a constructor. The main reason I left it in was because `launchProjectFor()` needs the delegate/data source's address at project launch, because the JB front end does not have a setting or checkbox to attach a data source to the project, not that I've found at least. So I was deploying, then making sure the launch project call executed properly, then calling `inititalize()`. But I guess we don't lose too much if `launchProjectFor()` fails. We'd still have to call it again, so gas cancels out there. There's an extra gas fee on the data source redeployment in the constructor version, and then we run the script again. So, trade an unlikely extra gas fee for a simpler contract/workflow. I'm on board.